utils/build/docker/ruby/parametric/server.rb

```diff
@@ -36,515 +38,826 @@
+  attr_accessor :span_id, :trace_id, :trace_flags, :trace_state, :remote
 
-    OtelFlushSpansReturn.new(success: success)
+  def initialize(span_id, trace_id, trace_flags, trace_state, remote)
+    @span_id = span_id
+    @trace_id = trace_id
+    @trace_flags = trace_flags
+    @trace_state = trace_state
+    @remote = remote
   end
 
-  def otel_flush_trace_stats(_otel_flush_trace_stats_args, _call)
-    OtelFlushTraceStatsReturn.new
+  def to_json(*_args)
+    {
+      span_id: @span_id,
+      trace_id: @trace_id,
+      trace_flags: @trace_flags,
+      trace_state: @trace_state,
+      remote: @remote
+    }.to_json
   end
+end
 
-  def stop_tracer(stop_tracer_args, _call)
-    Datadog.shutdown!
-    StopTracerReturn.new
+class OtelSetStatusArgs
+  attr_accessor :span_id, :code, :description
 
-    @otel_spans.clear
-    @dd_spans.clear
-    @dd_traces.clear
+  def initialize(params)
+    @span_id = params['span_id']
+    @code = params['code']
+    @description = params['description']
   end
+end
 
-  # The Ruby tracer holds spans on a per-Fiber basis.
-  # To allow for `#start_span`/`#finish_span` pairs to work seemly,
-  # the easiest way is to ensure all calls to this server execute in a single context.
-  #
-  # Because Fibers cannot be resumed across different threads, and this gRPC
-  # server handles each request in a different thread, we are using the next best thing,
-  # Threads, to ensure we are executing all requests to this server in a single thread.
-  # This allows `ddtrace` to handle trace and span context natively.
-  def initialize
-    super
+class OtelSetStatusReturn
+  def to_json(*_args)
+    {}.to_json
+  end
+end
 
-    @request_queue = Queue.new
-    @return_queue = Queue.new
+class OtelSetNameArgs
+  attr_accessor :span_id, :name
 
-    @thread = Thread.new do
-      loop do
-        m, args = @request_queue.pop
-        ret = public_send(m, *args)
-        @return_queue.push(ret)
-      rescue StandardError => e
-        @return_queue.push(e)
-      end
+  def initialize(params)
+    @span_id = params['span_id']
+    @name = params['name']
+  end
+end
+
+class OtelSetNameReturn
+  def to_json(*_args)
+    {}.to_json
+  end
+end
+
+class OtelSetAttributesArgs
+  attr_accessor :span_id, :attributes
+
+  def initialize(params)
+    @span_id = params['span_id']
+    @attributes = params['attributes']
+  end
+end
+
+class OtelSetAttributesReturn
+  def to_json(*_args)
+    {}.to_json
+  end
+end
+
+def get_ddtrace_version
+  Gem::Version.new(Datadog::VERSION)
+end
+
+def extract_http_headers(headers)
+  if Datadog::Tracing::Contrib::HTTP.respond_to?(:extract)
+    Datadog::Tracing::Contrib::HTTP.extract(headers.to_h)
+  else
+    Datadog::Tracing::Contrib::HTTP::Distributed::Propagation.new.extract(headers.to_h)
+  end
+end
+
+# OTel system tests provide times in microseconds, but Ruby OTel
+# measures time in seconds (Float).
+def otel_correct_time(microseconds)
+  if microseconds.nil? || microseconds == 0
+    nil
```

Comment:
Shouldn't we return microseconds (nil or 0) in case if it's `nil` or `0`

```suggestion
  if microseconds.nil? || microseconds == 0
    microseconds
```

Reply:
We didn't before in the gRPC implementation, but sure makes sense.